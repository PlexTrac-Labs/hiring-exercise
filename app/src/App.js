import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import axios from "axios";

import logo from "./logo.svg";
import "./App.css";

//Importing pages
import NavBar from "./components/NavBar";
import Login from "./pages/Login";
import Profile from "./pages/Profile";
import UserList from "./pages/UserList";

export const HOST = process.env.host || "localhost";
export const PORT = process.env.port || 3000;
export const BACKEND = process.env.backend || "http://localhost:5000";

axios.defaults.baseURL = BACKEND; // the prefix of the URL
axios.defaults.headers.get["Accept"] = "application/json"; // default header for all get request
axios.defaults.headers.post["Accept"] = "application/json"; // default header for all POST request

function App() {
  return (
    <Router>
      <NavBar />
      <Routes>
        <Route exact path="/" exact element={<Login />} />
        <Route path="/login" element={<Login />} />
        <Route path="/profile" element={<Profile />} />
        <Route path="/userlist" element={<UserList />} />
      </Routes>
    </Router>

    // <>
    //   <Login />
    // </>

    // <>
    //   <form className="login">
    //     <label>
    //       Name:
    //       <input type="text" name="name" />
    //     </label>
    //     <lable>
    //       Password:
    //       <input type="text" name="password" />
    //     </lable>
    //     <input type="submit" />
    //   </form>
    // </>

    //Old Code
    // <div className="App">
    //   <header className="App-header">
    //     <img src={logo} className="App-logo" alt="logo" />
    //     <p>
    //       Edit <code>src/App.js</code> and save to reload.
    //     </p>
    //     <a
    //       className="App-link"
    //       href="https://reactjs.org"
    //       target="_blank"
    //       rel="noopener noreferrer"
    //     >
    //       Learn React
    //     </a>
    //   </header>
    // </div>
  );
}

export default App;
