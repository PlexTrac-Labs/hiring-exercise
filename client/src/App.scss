.App {
  text-align: center;
  display: flex;
  flex-direction: column;

  .top-bar {
    width: 100%;
    .sign-out-btn {
      color: white !important;
      background-color: red !important;
      display: inline;
      float: right;
      margin: 8px;
    }

    .home-btn {
      color: white !important;
      display: inline;
      float: left;
      margin: 8px;
    }
  }

  .password-reset-container {
    position: fixed;
    bottom: 0;
    right: 0;
    margin: 8px;
  }
}
