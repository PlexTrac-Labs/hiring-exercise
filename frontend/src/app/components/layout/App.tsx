import { useEffect } from 'react';
import CommonHeader from '../commonHeader/CommonHeader';
import RouteMap from '../../routes';
import { useStore } from '../../stores/store';

function App() {
  const { sessionStore } = useStore();

  useEffect(() => {
    if (sessionStore.token) {
      sessionStore.getCurrentUser().finally(() => sessionStore.setAppLoaded());
    } else {
      sessionStore.setAppLoaded();
    }
  }, [sessionStore]);

  return (
    <div>
      <CommonHeader />
      <RouteMap />
    </div>
  );
}

export default App;
