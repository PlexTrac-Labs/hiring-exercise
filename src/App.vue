<template>
  <div class="container-fluid">
    <div class="border border-dark mb-5">
      <h1>User Management System</h1>
    </div>
    <!-- <Home /> -->
    <router-view></router-view>
  </div>
</template>

<script>
//import Home from "./components/Home.vue";

export default {
  name: "App",
  components: {
    //Home,
  }
};
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
  background-color: lightgrey;
}
html {
  background-color: lightgrey;
}
</style>
