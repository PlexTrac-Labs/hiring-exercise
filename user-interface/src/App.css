body {
    background-color: #121212;
}

/* Navigation Header */
nav {
    display: flex;
    justify-content: flex-end;
    align-items: flex-end;
    padding-top: 10px;
}

    nav .navLink {
        color: white;
        margin: 0 15px;
    }
