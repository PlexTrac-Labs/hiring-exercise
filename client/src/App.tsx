import React, { useState } from "react";
import "./App.scss";
import { BrowserRouter, Link, Route, Switch } from "react-router-dom";
import { LoginComponent } from "./components/Login/Login";
import {
  AuthService,
  IAuthService
} from "./services/Authentication/Authentication";
import { UsersList } from "./components/UsersList/UserList";
import { User } from "./models/User/User";
import { IUserService, UserService } from "./services/User/User";
import { UpdateUser } from "./components/UpdateUser/UpdateUser";
import { Button } from "@material-ui/core";
import { UserDetails } from "./components/UserDetails/UserDetails";
import { PasswordReset } from "./components/PasswordReset/PasswordReset";
import { Signout } from "./components/Signout/Signout";

export const apiBaseUrl: string = "http://localhost:5000";

export interface IContext {
  authService: IAuthService;
  userService: IUserService;
  user?: User;
  getAccessToken(): string;
  setAccessToken(token: string): void;
}

const Context: IContext = {
  authService: new AuthService(apiBaseUrl),
  userService: new UserService(apiBaseUrl),
  getAccessToken(): string {
    return sessionStorage.getItem("accessToken") ?? "";
  },
  setAccessToken(token: string): void {
    sessionStorage.setItem("accessToken", token);
  }
};

export const Ctx = React.createContext<IContext>({} as IContext);

const App: React.FC = () => {
  document.body.style.backgroundColor = "dodgerblue";
  const [token, setToken] = useState<string>(Context.getAccessToken());

  if (!token) {
    return <LoginComponent setToken={setToken} />;
  }

  return (
    <div className="App">
      <Ctx.Provider value={Context}>
        <BrowserRouter>
          <div className="top-bar">
            <Link to="/">
              <Button className="home-btn" variant="outlined">
                Home
              </Button>
            </Link>
            <Link to="/signout">
              <Button className="sign-out-btn" variant="contained">
                Sign out
              </Button>
            </Link>
          </div>
          <Switch>
            <Route exact path="/">
              <UsersList />
            </Route>
            <Route exact path="/login">
              <LoginComponent setToken={setToken} />
            </Route>
            <Route exact path="/updateuser/:id">
              <UpdateUser />
            </Route>
            <Route exact path="/user/:id">
              <UserDetails></UserDetails>
            </Route>
            <Route exact path="/passwordreset">
              <PasswordReset></PasswordReset>
            </Route>
            <Route exact path="/signout">
              <Signout setToken={setToken}></Signout>
            </Route>
          </Switch>
          <div className="password-reset-container">
            <p>
              Click <Link to="/passwordreset">here</Link> to reset password
            </p>
          </div>
        </BrowserRouter>
      </Ctx.Provider>
    </div>
  );
};

export default App;
