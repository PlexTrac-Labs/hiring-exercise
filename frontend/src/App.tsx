import { styled } from "@mui/system";
import React from "react";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import "./App.css";
import pages from "./pages";

const AppDiv = styled("div")({
  minHeight: "100vh",
});

function App() {
  return (
    <BrowserRouter>
      <AppDiv>
        <div>
          <Routes>
            {pages.map((page) => {
              return (
                <Route
                  key={page.route}
                  path={page.route}
                  element={<page.comp />}
                />
              );
            })}
          </Routes>
        </div>
      </AppDiv>
    </BrowserRouter>
  );
}

export default App;
