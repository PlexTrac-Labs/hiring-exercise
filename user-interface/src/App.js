import React from 'react';
import { Route, Switch } from 'react-router-dom';
import './App.css';
import Login from './components/Login';
import UserList from './components/UserList';

function getToken() {
    const tokenString = sessionStorage.getItem('token');
    // TODO: Validate that the token is not expired before returning the string. If expired, make sure we redirect to the Login page
    return tokenString;
}

export default function App() {
    //const classes = useStyles;

    function setToken(userToken) {
        sessionStorage.setItem('token', userToken.auth_token);
        window.location.reload();
    }

    // TODO: Make sure that this redirects to the user list when the user is authenticated, so that they don't have to refresh
    // Because I'm returned a Login component here instead of the Routing, pushing a new value to the history doesn't redirect since it can't route
    // I modified setToken to just reload() the window, which works. Not sure if there are any down-sides to this, but there may be a better solution?
    const token = getToken();
    if (!token) {
        return <Login setToken={setToken} />
    }

    return (
        <div>
            <Switch>
                <Route path="/UserList" component={UserList} />
                <Route path="/Login" component={Login} />
                <Route path="/" component={UserList} />
            </Switch>
        </div>
    );
};