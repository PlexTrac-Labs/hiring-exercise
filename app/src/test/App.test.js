import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "../App";

//Test to make sure the App, entry point to everything works.
it("renders correctly", async () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
