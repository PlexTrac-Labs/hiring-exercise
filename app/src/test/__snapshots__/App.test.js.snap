// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="navigation"
  >
    <nav
      className="navbar navbar-expand navbar-dark bg-dark"
    >
      <div
        className="container"
      >
        <a
          aria-current="page"
          className="navbar-brand active"
          href="/"
          onClick={[Function]}
        >
          PlexTrac Hiring Exercise
        </a>
        <div>
          <ul
            className="navbar-nav ml-auto"
          >
            <li
              className="nav-item"
            >
              <a
                aria-current="page"
                className="nav-link active"
                href="/"
                onClick={[Function]}
              >
                Log In
                <span
                  className="sr-only"
                >
                  (For now)
                </span>
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/profile"
                onClick={[Function]}
              >
                Profile
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/userlist"
                onClick={[Function]}
              >
                User List
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/logout"
                onClick={[Function]}
              >
                Log Out
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>,
  <div
    className="row align-items-center g-lg-5 py-5 col-md-10 mx-auto col-lg-5"
  >
    <form
      className="pform-signin login"
      onSubmit={[Function]}
    >
      <div
        className="form-floating"
      >
        <input
          className="form-control"
          id="UserNameInput"
          name="name"
          required={true}
          type="text"
        />
        <label
          htmlFor="UserNameInput"
        >
          User Name
        </label>
      </div>
      <div
        className="form-floating"
      >
        <input
          className="form-control"
          id="passwordInput"
          name="password"
          required={true}
          type="text"
        />
        <label
          htmlFor="passwordInput"
        >
          Password
        </label>
      </div>
      <button
        className="w-100 btn btn-lg btn-primary"
        type="submit"
      >
        Log In
      </button>
    </form>
  </div>,
]
`;
